src/global/tracking/CKFSeededTracking_factory.cc

```diff
@@ -0,0 +1,109 @@
+// Created by Dmitry Romanov
+// Subject to the terms in the LICENSE file found in the top-level directory.
+//
+
+
+#include <Acts/Propagator/Navigator.hpp>
+#include "CKFSeededTracking_factory.h"
+#include "extensions/spdlog/SpdlogExtensions.h"
+#include <JANA/JEvent.h>
+#include "services/geometry/acts/ACTSGeo_service.h"
+#include "services/log/Log_service.h"
+#include "extensions/string/StringHelpers.h"
+#include <services/geometry/dd4hep/JDD4hep_service.h>
+
+void eicrecon::CKFSeededTracking_factory::Init() {
+    auto app = GetApplication();
+
+    // This prefix will be used for parameters
+    std::string plugin_name = eicrecon::str::ReplaceAll(GetPluginName(), ".so", "");
+    std::string param_prefix = plugin_name+ ":" + GetTag();
+
+    // Initialize input tags
+    InitDataTags(param_prefix);
+
+    // Initialize logger
+    InitLogger(param_prefix, "info");
+
+    // Get ACTS context from ACTSGeo service
+    auto acts_service = app->GetService<ACTSGeo_service>();
+    auto dd4hp_service = app->GetService<JDD4hep_service>();
+
+
+    // Algorithm configuration
+    auto cfg = GetDefaultConfig();
+    app->SetDefaultParameter(param_prefix + ":EtaBins", cfg.m_etaBins, "Eta Bins for ACTS CKF tracking reco");
+    app->SetDefaultParameter(param_prefix + ":Chi2CutOff", cfg.m_chi2CutOff, "Chi2 Cut Off for ACTS CKF tracking");
+    app->SetDefaultParameter(param_prefix + ":NumMeasurementsCutOff", cfg.m_numMeasurementsCutOff, "Number of measurements Cut Off for ACTS CKF tracking");
+
+    // Initialize algorithm
+    m_tracking_algo.applyConfig(cfg);
+    m_tracking_algo.init(acts_service->actsGeoProvider(), m_log);
+}
+
+void eicrecon::CKFSeededTracking_factory::ChangeRun(const std::shared_ptr<const JEvent> &event) {
+
+}
+
+void eicrecon::CKFSeededTracking_factory::Process(const std::shared_ptr<const JEvent> &event) {
+    // Now we check that user provided an input names
+    std::string input_tag = GetInputTags()[0];
```

Comment:
Here's where I would modify (in `CKFTracking_factory.cc`) to:
```suggestion
    std::string track_params = GetInputTags()[0];
    std::string source_links = GetInputTags()[1];
```
and then in the plugin cc use
```
    app->Add(new JChainFactoryGeneratorT<TrackParamSeeding_factory>(
            {"CentralTrackSeedingResults"}, "SeededTrackParams"));
    app->Add(new JChainFactoryGeneratorT<CKFTracking_factory>(
            {"SeededTrackParams", "CentralTrackerSourceLinker"}, "CentralCKFSeededTrajectories"));
```

Reply:
Do you mean instead of this whole file?

Reply:
Yes. The suggestion illustrates the location of the change suggested to the existing code to be reused instead of adding a copy.

Reply:
I deleted `CKFSeededTracking_factory` and made the changes to `CKFTracking_factory` and the `tracking.cc` plugin.

Reply:
In order for the `CentralCKFSeededTrajectories `to be used in the simulation campaign, I will need to add two additional lines to the `tracking.cc` plugin:

```
app->Add(new JChainMultifactoryGeneratorT<TrackingResult_factory>(
            "CentralTrackingParticles",                       // Tag name for multifactory
            {"CentralCKFSeededTrajectories"},                       // eicrecon::TrackingResultTrajectory
            {"outputSeededParticles",                               // edm4eic::ReconstructedParticle
             "outputSeededTrackParameters"},                        // edm4eic::TrackParameters
            app)); 

app->Add(new JChainMultifactoryGeneratorT<ParticlesWithTruthPID_factory>(
            "ChargedParticlesWithAssociations",                // Tag name for multifactory
            {"MCParticles",                                    // edm4hep::MCParticle
            "outputSeededTrackParameters"},                          // edm4eic::TrackParameters
            {"ReconstructedSeededChargedParticles",                  //
             "ReconstructedSeededChargedParticleAssociations"       // edm4eic::MCRecoParticleAssociation
            },
            app  // TODO: Remove me once fixed
            ));
```

Reply:
I added this.